src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -129,71 +143,94 @@ function PartUsageDialog(props) {
-        updateFormDataProperty(FORM.ASSET, '');
-        updateFormDataProperty(FORM.ASSET_DESC, '');
-        updateFormDataProperty(FORM.BIN, '');
-        updateFormDataProperty(FORM.PART, '');
-        updateFormDataProperty(FORM.PART_DESC, '');
-        // Reset bin list
-        setBinList([]);
+    const resetFieldWithDesc = (fieldValueKey, fieldDescKey) => {
+        updateFormDataProperty(fieldValueKey, '');
+        updateFormDataProperty(fieldDescKey, '');
+    }
+
+    const resetFieldWithList = (fieldValueKey, listSetterFunction) => {
+        updateFormDataProperty(fieldValueKey, '');
+        listSetterFunction([]);
+    }
+
+    const resetFormTransactionLinesAndRelatedStates = () => {
+        // Reset 'transactionlines' proprieties (including associated options lists)
+        resetFieldWithDesc(FORM.ASSET, FORM.ASSET_DESC);
+        resetFieldWithDesc(FORM.PART, FORM.PART_DESC);
+        resetFieldWithList(FORM.BIN, setBinList);
+        resetFieldWithList(FORM.LOT, setLotList);
+
+        // Reset other
+        setUoM('');
+        setIsTrackedByAsset(false);
     };
 
     const handleTransactionChange = () => {
-        resetFormTransactionLinesAndBinListStates();
+        resetFormTransactionLinesAndRelatedStates();
     };
 
     const handleStoreChange = () => {
-        resetFormTransactionLinesAndBinListStates();
+        resetFormTransactionLinesAndRelatedStates();
     };
 
     /* This function handles at least 3 cases:
-     * 1) The asset ID input is cleared.
-     * 2) The user searches for an asset ID directly (without having selected a part),
-     *    so we have to update the remaining state concerning part and bin.
-     * 3) There was already a part selected so we only need to update the bin state.
+     * 1) The asset ID field is cleared, is solely whitespace or is falsy.
+     * 2) The user searches for an asset ID without having selected a part or by selecting it from the history
+     *    (so we need to update the related state: part, bin and lot).
+     * 3) There was already a part selected (so we only need to update the bin and lot states).
      */
     const handleAssetChange = async (assetIDCode) => {
-        // Asset input cleared
-        if (assetIDCode?.trim() === '') {
-            updateFormDataProperty(FORM.ASSET, '');
-            updateFormDataProperty(FORM.ASSET_DESC, '');
-            updateFormDataProperty(FORM.BIN, '');
-            setBinList([]);
+        const { transactionType, partCode } = formData;
+
+        // Reset related fields
+        updateFormDataProperty(FORM.BIN, '');
+        if (transactionType === issueType) {
+            // In a return transaction the bin list is loaded when a part is selected so no need to reset
+            resetFieldWithList(FORM.BIN, setBinList);
+        }
+        resetFieldWithList(FORM.LOT, setLotList);
+
+        // Asset field is definitely not an asset code
+        if (!assetIDCode || assetIDCode?.trim() === '') {
```

Comment:
no need to optional chain, we already know that there is an assetIDCode because of short-circuit